### PR TITLE
Instead of adding additional logins via php api - this can perfectly …

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…be done via the config as well

## Screenshots (if appropriate):

bildschirmfoto von 2017-04-10 10-14-27

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: core/Controller/LoginController.php [hunk 1]
```diff
@@ -146,7 +146,12 @@ public function showLoginForm($user, $redirect_url, $remember_login) {
 			}
 		}
 
-		$parameters['alt_login'] = OC_App::getAlternativeLogIns();
+		$altLogins = OC_App::getAlternativeLogIns();
+		$altLogins2 = $this->config->getSystemValue('login.alternatives');
```
Comment (PVince81) [inline core/Controller/LoginController.php]: please add to `config.sample.php`

**Comment by PVince81** [comment]
:+1:

**Comment by PVince81** [comment]
Failing tests:
```
10:39:16 1) Tests\Core\Controller\LoginControllerTest::testShowLoginFormWithPasswordResetOption with data set #0 (true, true)
10:39:16 OCP\IConfig::getSystemValue('login.alternatives', '') was not expected to be called more than once.
10:39:16 
10:39:16 /var/lib/jenkins/workspace/owncloud-core_core_PR-27607-WK4NWSBUHCZGNS5OJCGIPJE5RNXHDVGK7VCLROJOBTRLQP6HBOYQ/core/Controller/LoginController.php:150
10:39:16 /var/lib/jenkins/workspace/owncloud-core_core_PR-27607-WK4NWSBUHCZGNS5OJCGIPJE5RNXHDVGK7VCLROJOBTRLQP6HBOYQ/tests/Core/Controller/LoginControllerTest.php:240
10:39:16 
10:39:16 2) Tests\Core\Controller\LoginControllerTest::testShowLoginFormWithPasswordResetOption with data set #1 (false, false)
10:39:16 OCP\IConfig::getSystemValue('login.alternatives', '') was not expected to be called more than once.
10:39:16 
10:39:16 /var/lib/jenkins/workspace/owncloud-core_core_PR-27607-WK4NWSBUHCZGNS5OJCGIPJE5RNXHDVGK7VCLROJOBTRLQP6HBOYQ/core/Controller/LoginController.php:150
10:39:16 /var/lib/jenkins/workspace/owncloud-core_core_PR-27607-WK4NWSBUHCZGNS5OJCGIPJE5RNXHDVGK7VCLROJOBTRLQP6HBOYQ/tests/Core/Controller/LoginControllerTest.php:240
10:39:16 
10:39:16 3) Tests\Core\Controller\LoginControllerTest::testShowLoginFormForUserNamedNull
10:39:16 OCP\IConfig::getSystemValue('login.alternatives', '') was not expected to be called more than once.
10:39:16 
10:39:16 /var/lib/jenkins/workspace/owncloud-core_core_PR-27607-WK4NWSBUHCZGNS5OJCGIPJE5RNXHDVGK7VCLROJOBTRLQP6HBOYQ/core/Controller/LoginController.php:150
10:39:16 /var/lib/jenkins/workspace/owncloud-core_core_PR-27607-WK4NWSBUHCZGNS5OJCGIPJE5RNXHDVGK7VCLROJOBTRLQP6HBOYQ/tests/Core/Controller/LoginControllerTest.php:279
```

File: config/config.sample.php [hunk 2]
```diff
@@ -203,6 +203,15 @@
 'token_auth_enforced' => false,
 
 /**
+ * Allows to specify additional login buttons on the logon screen for e.g. SSO integration
+ *	'login.alternatives' => [
+ *    ['href' => '/sso-saml', 'name' => 'Shibboleth SSO', 'img' => '/img/shibboleth.png'],
+ *    ['href' => '/sso-adfs', 'name' => 'ADFS'],
```
Comment (butonic) [inline config/config.sample.php]: Very bad example: ADFS and Shibboleth are both doing SAML. The use cases out there are showing buttons for different IdPs. The links then go directly to the idp. on oic side the same `/login/saml` url is protected by mod_shib. It is configured to handle multiple idps with dedicated SessionInitiators, see https://wiki.shibboleth.net/confluence/display/CONCEPT/IdPDiscovery#IdPDiscovery-AttheApplication

```html
<a href="https://www.testshib.org/Shibboleth.sso/ProtectNetwork?target=https%3A%2F%2Fwww.testshib.org%2Ftestshib-two%2Fauth-pages%2Fauth.jsp">
   <img src="images/PN_sign-in.gif" alt="Login with ProtectNetwork" border="0">
</a>
 
<a href="https://www.testshib.org/Shibboleth.sso/OpenIdP.org?target=https%3A%2F%2Fwww.testshib.org%2Ftestshib-two%2Fauth-pages%2Fauth.jsp">
   <img src="images/openidp.png" alt="Login with OpenIdP.org" border="0">
</a>
```

Comment (butonic) [inline config/config.sample.php]: make it
```
* Allows to specify additional login buttons on the logon screen for e.g. SSO integration
*	'login.alternatives' => [
*    ['href' => 'https://www.testshib.org/Shibboleth.sso/ProtectNetwork?target=https%3A%2F%2Fmy.owncloud.tld%2Flogin%2Fsso-saml%2F', 'name' => 'ProtectNetwork', 'img' => '/img/PN_sign-in.gif'],
*    ['href' => 'https://www.testshib.org/Shibboleth.sso/OpenIdP.org?target=https%3A%2F%2Fmy.owncloud.tld%2Flogin%2Fsso-saml%2F', 'name' => 'OpenIdP.org'], 'img' => '/img/openidp.png'],
```

Comment (DeepDiver1975) [inline config/config.sample.php]: I'm happy to change this to something more reasonable @butonic

File: config/config.sample.php [hunk 3]
```diff
@@ -203,6 +203,15 @@
 'token_auth_enforced' => false,
 
 /**
+ * Allows to specify additional login buttons on the logon screen for e.g. SSO integration
+ *	'login.alternatives' => [
+ *	'login.alternatives' => [
```
Comment (butonic) [inline config/config.sample.php]: 207 === 208

File: config/config.sample.php [hunk 4]
```diff
@@ -203,6 +203,15 @@
...
+ *    ['href' => 'https://www.testshib.org/Shibboleth.sso/ProtectNetwork?target=https%3A%2F%2Fmy.owncloud.tld%2Flogin%2Fsso-saml%2F', 'name' => 'ProtectNetwork', 'img' => '/img/PN_sign-in.gif'],
+ *    ['href' => 'https://www.testshib.org/Shibboleth.sso/OpenIdP.org?target=https%3A%2F%2Fmy.owncloud.tld%2Flogin%2Fsso-saml%2F', 'name' => 'OpenIdP.org'], 'img' => '/img/openidp.png'],
+ * *  ]
```
Comment (butonic) [inline config/config.sample.php]: * *

File: config/config.sample.php (hunk 3 above)
Comment (DeepDiver1975) [inline config/config.sample.php]: :wtf?:

File: config/config.sample.php [hunk 5]
```diff
@@ -203,6 +203,14 @@
...
 
 /**
+ * Allows to specify additional login buttons on the logon screen for e.g. SSO integration
+ *  'login.alternatives' => [
+ *    ['href' => 'https://www.testshib.org/Shibboleth.sso/ProtectNetwork?target=https%3A%2F%2Fmy.owncloud.tld%2Flogin%2Fsso-saml%2F', 'name' => 'ProtectNetwork', 'img' => '/img/PN_sign-in.gif'],
+ *    ['href' => 'https://www.testshib.org/Shibboleth.sso/OpenIdP.org?target=https%3A%2F%2Fmy.owncloud.tld%2Flogin%2Fsso-saml%2F', 'name' => 'OpenIdP.org'], 'img' => '/img/openidp.png'],
+ *  ]
+ */
+'login.alternatives' => [],
+/**
```
Comment (butonic) [inline config/config.sample.php]: missing newline for clarity

**Comment by PVince81** [comment]
With only one option it doesn't look good, needs some centering:
alt-login

**Comment by DeepDiver1975** [comment]
> With only one option it doesn't look good, needs some centering:

maybe @felixheidecke can apply some css magic

**Comment by PVince81** [comment]
I'll just add "text-align: center". As easy as that:
buttonzzz

**Comment by PVince81** [comment]
Here you go: https://github.com/owncloud/core/pull/27607/commits/67486aa6006b662518aa9d2229e0036df2dbe376

That's just CSS magic level 0

**Pull request merged**